pkg/shared/service.go

```diff
@@ -18,6 +22,103 @@ func NewServiceHandler() *ServiceHandler {
 
 func (sh *ServiceHandler) Handle(ctx context.Context, event sdk.Event) error {
 	fmt.Println("handling object ", event.Object.GetObjectKind().GroupVersionKind().String())
+
+	sharedService := event.Object.(*v1alpha1.SharedService)
+	sharedServiceCopy := sharedService.DeepCopy()
+
+	if sharedService.Spec.ServiceType != strings.ToLower(v1alpha1.KeycloakKind) {
+		return nil
+	}
+
+	if event.Deleted {
+		return nil
+	}
+
+	if sharedService.GetDeletionTimestamp() != nil {
+		return sh.finalizeSharedService(sharedServiceCopy)
+	}
+
+	switch sharedService.Status.Phase {
+	case v1alpha1.SSPhaseNone:
+		sh.initSharedService(sharedServiceCopy)
+	case v1alpha1.SSPhaseAccepted:
+		sh.createKeycloaks(sharedServiceCopy)
+	case v1alpha1.SSPhaseComplete:
+		//ToDo What to do here?
```

Comment:
I'm assuming this operator is supposed to handle the creation of all keycloak resources so should be i assume monitoring the current state all the time (instances running vs number required) to make sure it isn't supposed to be creating more.

Reply:
yes, exactly. If the number required is different to the number running this should be creating new KeyCloak objects, or looking for unused keycloak objects to delete.